impls/src/adapters/http.rs

```diff
@@ -144,32 +140,24 @@ impl WalletCommAdapter for HTTPWalletCommAdapter {
 
 		Ok(slate)
 	}
+}
 
-	fn send_tx_async(&self, _dest: &str, _slate: &Slate) -> Result<(), Error> {
-		unimplemented!();
-	}
-
-	fn receive_tx_async(&self, _params: &str) -> Result<Slate, Error> {
-		unimplemented!();
-	}
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+pub struct SchemeNotHttp;
 
-	fn listen(
-		&self,
-		_params: HashMap<String, String>,
-		_config: WalletConfig,
-		_passphrase: &str,
-		_account: &str,
-		_node_api_secret: Option<String>,
-	) -> Result<(), Error> {
-		unimplemented!();
+impl Into<Error> for SchemeNotHttp {
+	fn into(self) -> Error {
+		let err_str = format!("url scheme must be http",);
+		ErrorKind::GenericError(err_str).into()
 	}
 }
 
-pub fn post<IN>(url: &str, api_secret: Option<String>, input: &IN) -> Result<String, api::Error>
+pub fn post<IN>(url: &Url, api_secret: Option<String>, input: &IN) -> Result<String, api::Error>
```

Comment:
`&str` is changed to `&Url` in pursuit of [type safety](https://rust-lang-nursery.github.io/api-guidelines/type-safety.html).